pkg/node/manager/manager.go

```diff
@@ -167,6 +182,21 @@ func (m *manager) AddNode(nodeName string) error {
 	return nil
 }
 
+func (m *manager) CreateCNINodeIfNotExisting(node *v1.Node) error {
+	if cniNode, err := m.wrapper.K8sAPI.GetCNINode(
+		types.NamespacedName{Name: node.Name},
+	); err != nil {
+		if apierrors.IsNotFound(err) {
+			m.Log.Info("Will create a new CNINode", "CNINodeName", node.Name)
+			return m.wrapper.K8sAPI.CreateCNINode(node)
+		}
+		return err
+	} else {
```

Comment:
AFAICT, this else doesn't do anything, since all other branches shortcircuit.

Reply:
We don't need `else` but I think if...else should also work. This else is for the if check on `err != nil`